### PR TITLE
Add code excerpts to new type promotion failures

### LINKED ISSUE #5350: [3.2] Add code excerpts to non-promotion-reasons
Lots of new code examples in #5246, should convert them all to code excerpts. 

- `{:.bad}` snippets don't get an excerpt (just going to cause errors)
- Right now, all the examples share generic class names and field names... there might be a way to section off the code for the snippets but we're either going to have to change a lot of the identifiers to fit them in the same file, or make a few new files. TBD

### PR DESCRIPTION
**Pull request by MaryaBelanger**
Excerpts added in #5246

Fixes #5350

### CODE REVIEW

**Comment by MaryaBelanger** [comment]
I'm still not sure how to implement the excerpts for the `{:.bad}` examples, tbd. Maybe the file just has a bunch of `ignore_for_file` conditions? 

I know there's a way to print the actual error from the excerpts too (I think the [Fixing common type problems](https://dart.dev/guides/language/sound-problems) page does it), so tbd on that too.

File: src/tools/non-promotion-reasons.md [hunk 1]
```diff
@@ -343,15 +346,16 @@ f(C c) {
 Assign the getter to a local variable:
 
 {:.good}
+<?code-excerpt "non_promotion/lib/non_promotion.dart (not-field)" replace="/final.*/[!$&!]/g"?>
 ```dart
 import 'dart:math';
-
-abstract class C {
+// ···
```
Comment (MaryaBelanger) [inline src/tools/non-promotion-reasons.md:352]: Not huge fan of how this is automatically added anywhere the `docregion`s end/restart in the excerpts file

Comment (parlough) [reply src/tools/non-promotion-reasons.md:352]: You can disable this by adding `<?code-excerpt plaster="none"?>` to the top of the file or in the excerpt you don't want to have it:

```
<?code-excerpt "file.dart (region)" plaster="none"?>
```

**Comment by atsansone** [comment]
@MaryaBelanger : Once you get this PR updated, I'll take another look.

**Comment by atsansone** [comment]
@MaryaBelanger : This was last touched by you in 2023. Do you think you can come back to this or should someone else pick it up?

**Comment by parlough** [comment]
I took this one over and will try to get back to it this week.

**Review by parlough — APPROVED** [review]
Thanks for doing this @MaryaBelanger, looks great to me!

I pulled out some of the copy fixes and clean up out into https://github.com/dart-lang/site-www/pull/5859, then rebased on top of that cleanup, and finally fixed the few remaining errors.

**Pull request merged**